Engine/src/Engine/Scene.cpp

```diff
@@ -30,6 +30,27 @@ namespace Engine
 		glfwSwapInterval(1);
 		glClearColor(0.1f, 0.1f, 0.1f, 1);
 
+		//initialize the window for UI
+		UserInterface::initialize(m_window);
+
+		const int menuHeight = 18;
+
+		//set position for the explorer UI element
+		m_explorerPanel.setPosition(glm::uvec2(0, menuHeight));
```

Comment:
I would suggest all of these should be moved to their own method, such as "setEditorWindow" or something similar, for example

Reply:
Similarly to my other comment, we also probably want a specific onEditorCreate() method

Reply:
so something like explorerPanel.setEditorWindow(position, dimension)?

Reply:
Thinking about it better you probably don't need a method for these, just define everything from ImGui in a separe create method (suggested name onEditorCreate(); That way we can make sure to not call the relevant editor function when an executable of the game is made.

Reply:
Created initializeUI()